### ISSUE TITLE
Install on MacOS fails because MKL is missing

### ISSUE THREAD

**Issue by ArchRobison** [rprechelt/Vectorize.jl#12]
I tried to install `Vectorize.jl` on my Mac (running MacOS El Capitan), and the build failed because MKL was missing.  Shouldn't it have just omitted MKL from consideration since it was missing?  See [here](https://gist.github.com/ArchRobison/c09755abe8704c2f910fd14cafda38ad#file-session-txt) for transcript.

**Comment by ArchRobison** [comment]
The install did succeed after I installed MKL.

**Comment by rprechelt** [comment]
You're correct - it should just not load/use any MKL instructions - I'll take a look at why the MKL detection is failing and get back to you ASAP. Thanks for trying it out!

**Comment by rprechelt** [comment]
This is strange - the error is caused by the lack of `libmkl_intel_thread.dylib`. However, `libmkl_intel_thread.dylib` is only loaded during MKL initialization which is guarded with a `if (Libdl.find_library(["libmkl_rt"], ["/opt/intel/mkl/lib"]) != ""` statement. Were the MKL files on your system but not installed (added to `PATH` etc.) such that `Libdl` may have been able to find the files?
